### PR TITLE
HPCC-20114 Don't allow service binding to https with no certificate

### PR DESCRIPTION
**Pull request by kenrowland**
Add requiredIf to HTTPS relevant attributes marking them required if
https is the selected binding protocol for an ESP service.
Mark HTTPS deprecated certificate attributes.
Fix omitted merge changes.

Signed-off-by: Ken Rowland <kenneth.rowland@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: initfiles/componentfiles/configschema/xsd/esp.xsd [hunk 1]
```diff
@@ -95,7 +95,7 @@
                                         </xs:restriction>
                                     </xs:simpleType>
                                 </xs:attribute>
-                                <xs:attribute name="ldapServer" type="xs:string" use="optional" hpcc:displayName="LDAP Server" hpcc:requiredIf=".[@method=('ldap','ldaps')]"
+                                <xs:attribute name="ldapServer" type="xs:string" use="optional" hpcc:displayName="LDAP Server" hpcc:requiredIf="../EspBinding[@method=('ldap','ldaps')]"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/esp.xsd]: "Authentication/@ldapServer" and "Authentication/@method" seem to be at the same level, and we might not need to go up and down to EspBinding.

File: initfiles/componentfiles/configschema/xsd/esp.xsd [hunk 2]
```diff
@@ -142,12 +142,24 @@
...
                                 <xs:attribute name="cipherList" type="xs:string" use="optional" default="ECDH+AESGCM:DH+AESGCM:ECDH+AES256:DH+AES256:ECDH+AES128:DH+AES:ECDH+3DES:DH+3DES:RSA+AESGCM:RSA+AES:RSA+3DES:!aNULL:!MD5"
                                               hpcc:displayName="Cipher List" hpcc:tooltip="Sets the ordered list of available ciphers for use by openssl.  See openssl documentation on ciphers for information on use and formatting."/>
                                 <xs:attribute name="passphrase" type="xs:string" use="optional" hpcc:modifiers="mask,verify,ignoreNoChange,encrypt" hpcc:displayName="Passphrase" hpcc:tooltip="The passphrase used to generate the private key" />
-                                <xs:attribute name="privateKeyFileName" type="xs:string" use="optional" default="privatekey.cer" hpcc:displayName="Private Key Filename" hpcc:tooltip="Name of destination file in which the private key will be written" />
+                                <xs:attribute name="privateKeyFileName" type="xs:string" use="optional" default="privatekey.cer" hpcc:displayName="Private Key Filename" hpcc:requiredIf="../EspBinding[@protocol='https']"
```
Comment (rpastrana) [inline initfiles/componentfiles/configschema/xsd/esp.xsd]: let's add a comment explaining why these are here and hidden

**Review by rpastrana — CHANGES_REQUESTED** [review]
@kenrowland a couple of comments

File: initfiles/componentfiles/configschema/xsd/esp.xsd (hunk 1 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/esp.xsd]: Well, it was correct in the previous revision. Not sure why I changed it. Corrected now.

File: initfiles/componentfiles/configschema/xsd/esp.xsd (hunk 2 above)
Comment (kenrowland) [reply initfiles/componentfiles/configschema/xsd/esp.xsd]: Comment added

**Review by kenrowland — COMMENTED** [review]
@rpastrana Comments addressed.

**Comment by kenrowland** [comment]
@rpastrana Please approve if latest changes are acceptable

**Review by rpastrana — APPROVED** [review]
@kenrowland

**Comment by kenrowland** [comment]
@richardkchapman please merge

**Pull request merged**